### DISCUSSION TITLE
Self-hosted runners action-runner folder

### DISCUSSION THREAD

**Discussion by jeremylaux** [community/community#25972] [Actions]
<p>I setup a self-hosted runner on a Windows 19 box and I’d like to use it for production deployment. (hopefully I’m using it properly)</p>
<p>I have my YML setup so that when a push goes into master, it runs a few cmds including an npm build and an a PM2 restart.</p>
<p>What I don’t understand is that my actions and code run within a new action-runner folder:</p>
<ul>
<li>C:\actions-runner\_work\app</li>
</ul>
<p>But my application is being served from another directory:</p>
<ul>
<li>C:\inetpub\app</li>
</ul>
<p>How to do I get my action changes to take place or reflect in the served location, not the _work directory?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The default working directory of your workflow when using your self-hosted runner is <code>C:\actions-runner\_work\your-repo-name\your-repo-name</code> . If you are using run scripts step, using the <code>working-directory</code> keyword, you can specify the working directory of where to run the command.</p>
<pre><code class="lang-auto">- name: Clean install dependencies and build
  run: |
    npm ci
    npm run build
    working-directory: C:\inetpub\app
</code></pre>

**Reply by Satishkaramana** [reply]
Very well explained. Thanks. How will this be specified using the default runners [windows-latest or ubuntu-latest]? Meaning, my applications .csproj file is located as: C:\actions-runner\_work\your-repo-name\your-repo-name\appweb\myappweb => myappweb.csproj. How will default runner find this path and run the dotnet build?

**Reply by OctopBP** [reply]
Just a short remark, because a faced it myself. The `working-directory: C:\inetpub\app` line shouldn't be aligned with run commands. I should be like this:

```
- name: Clean install dependencies and build
  run: |
    npm ci
    npm run build
  working-directory: C:\inetpub\app
```

**Reply by djjr44905-art** [reply]
> Le répertoire de travail par défaut de votre workflow lorsque vous utilisez votre exécuteur auto-hébergé est `C:\actions-runner\_work\your-repo-name\your-repo-name`. Si vous utilisez l'étape d'exécution de scripts, le `working-directory`mot-clé vous permet de spécifier le répertoire de travail où exécuter la commande.
> 
> ```
> - name: Clean install dependencies and build
>   run: |
>     npm ci
>     npm run build
>     working-directory: C:\inetpub\app
> ```

**Comment by jeremylaux** [comment]
<p>ahh, thank you, I’ll give this a try</p>

**Comment by jeremylaux** [comment]
<p>Hi Yanjingzhu, thanks for your help. I have it working at the proper working-directory now, but my entire application is also still in the default directory… Any idea if I can remove that?</p>
<p><code>C:\actions-runner\_work\your-repo-name\your-repo-name</code></p>

**Comment by Yanjingzhu** [comment]
<p>Sorry for the delay response. <a href="https://github.com/actions/checkout" rel="nofollow noopener">Actions/checkout</a>  action checks-out your repository under  <code>$GITHUB_WORKSPACE</code><br>
C:\actions-runner_work\your-repo-name\your-repo-name is the GitHub workspace of your workflow.<br>
There is a path input variable of checkout action to specify the path where you want to hold your repo.  Please take a note that the path is relative path under the default directory.</p>
<pre><code class="lang-auto">- uses: actions/checkout@v2
  with:
 # **Relative path** under $GITHUB_WORKSPACE to place the repository
    path: ''
</code></pre>
